src/action.js

```diff
@@ -10,68 +10,69 @@ const got = require('got')
 //https://github.com/broofa/mime
 const { getType } = require('mime')
 
-const tagName = core.getInput('tag_name', { required: true }),
-    bodyPath = core.getInput('body_path', { required: false }),
-    assetFile = core.getInput('asset_files', { require: false }),
-    draft = core.getInput('draft', { required: false }) === 'true',
-    prerelease = core.getInput('prerelease', { required: false }) === 'true',
+const 
+    tagName = core.getInput('tag_name', { required: false }),
     githubToken = process.env.GITHUB_TOKEN,
     octokit = GitHub.getOctokit(githubToken),
     context = GitHub.context,
     { owner: currentOwner, repo: currentRepo } = context.repo,
-    getRepo = core.getInput('repo', { required: false }) || currentOwner + "/" + currentRepo
-var releaseName = core.getInput('release_name', { required: false }).replace('refs/tags/', ''),
-    body = core.getInput('body', { required: false }),
-    tag = tagName.replace('refs/tags/', '')
-const
+    getRepo = core.getInput('repo', { required: false }) || currentOwner + "/" + currentRepo,
     owner = getRepo.match(/^[\s\w]+(?=\/)/g)[0],
     repo = getRepo.match(/[^\/][\d\w-]+$/g)[0]
 
-var assetArray = [],
-    bodyFileContent,
-    latestAsset,
-    uploadUrl
+var 
+    releaseName = core.getInput('release_name', { required: false }).replace('refs/tags/', ''),
+    body = core.getInput('body', { required: false }),
+    tag = tagName.replace('refs/tags/', '')
+    assetArray = [], bodyFileContent, releaseAsset, uploadUrl
 
-core.info(
-    format('tag_name:%s,release_name:%s,body:%s,body_path:%s,asset_files:%s,draft:%s,prerelease:%s,owner:%s,repo:%s',
-        tag, releaseName, body, bodyPath, assetFile, draft, prerelease, owner, repo))
+core.info(format('tag_name:%s, owner:%s, repo:%s', tagName, owner, repo))
 
 Main()
 
 async function Main() {
     try {
 
-        var LatestPromise
-        if (tag == 'inherit' && currentRepo != repo) {
-            LatestPromise = GetLatest()//update body releaseName
-        }
-
-        if (bodyPath !== '' && !!bodyPath) {
-            let bodyStr = fs.readFileSync(bodyPath, { encoding: 'utf8' })
-            core.info(bodyStr)
-            bodyFileContent = (body ? body + '\n' : '') + bodyStr
+        var ReleasePromise
+        if (tag == '') {
+            ReleasePromise = GetLatestRelease() // tag, releaseName, body
```

Comment:
can either of these calls fail?  if so, what happens?

Reply:
If the underlying call to octokit fails for either call, the action explicitly fails, as is expected of a well-behaved action:

```javascript
    tagRelease = await octokit.repos.getReleaseByTag({
        owner: currentOwner,
        repo: currentRepo,
        tag: tag
    }).catch(err => {
        core.setFailed(err)
        throw new Error(err)
    })
```

See lines 61 and 79.

The Tag version will 404 if the release doesn't exist, so ProcessRelease(...) should always have something to work on if the octokit call doesn't explicitly fail. The REST API path behind GetLatestRelease lists only 200 as an option, so it's possible we get an empty payload and ProcessRelease(...) fails on the attempt to destructure an empty payload. I think that's ok for two reasons:

-  The Tag version of the call is much more useful given the oddness we've seen with Latest release.
-  This is meant to run in a workflow in which a Release is created. If the preceding actions fail to create a Release the author of the workflow has other things to worry about.

I did think about just removing the Latest release path, but left it in, because it's already there and works for GitHub's definition of Latest.